packages/frontend-api/src/Model/Resolver/Transport/TransportsResolver.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Resolver\Transport;
+
+use Overblog\GraphQLBundle\Definition\Resolver\AliasedInterface;
+use Overblog\GraphQLBundle\Definition\Resolver\ResolverInterface;
+use Shopsys\FrameworkBundle\Model\Payment\PaymentFacade;
+use Shopsys\FrameworkBundle\Model\Transport\TransportFacade;
+
+class TransportsResolver implements ResolverInterface, AliasedInterface
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Transport\TransportFacade
+     */
+    protected $transportFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Payment\PaymentFacade
+     */
+    protected $paymentFacade;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Transport\TransportFacade $transportFacade
+     * @param \Shopsys\FrameworkBundle\Model\Payment\PaymentFacade $paymentFacade
+     */
+    public function __construct(TransportFacade $transportFacade, PaymentFacade $paymentFacade)
+    {
+        $this->transportFacade = $transportFacade;
+        $this->paymentFacade = $paymentFacade;
+    }
+
+    /**
+     * @return array
+     */
+    public function resolve(): array
+    {
+        return $transports = $this->transportFacade->getVisibleOnCurrentDomain($this->paymentFacade->getVisibleOnCurrentDomain());
```

Comment:
I'm a little confused. Why visible payments are resolved in this method, but visible transports are not resolved in PaymentFacade::getVisibleOnCurrentDomain?

Reply:
I think that reason there is that you will not display transport that has no visible payment

Reply:
I am just surprised that this is not same as the opposite :thinking: